.travis.yml

```diff
@@ -35,6 +35,9 @@ before_install:
 install:
     - travis_retry composer update ${COMPOSER_FLAGS} --prefer-dist --no-interaction
 
+before_script:
+    - if [[ "${TRAVIS_PHP_VERSION}" == "5.4" ]]; then composer remove slim/slim; fi
```

Comment:
I would move this to before install AND add --no-update flag to composer. That way we only have to run compose once.